File: Year-2023/LeetCode-September-challenge/MinimumDeletionstoMakeCharacterFrequenciesUnique.java
import java.util.*;

class Solution {
    public int minDeletions(String s) {
        int[] freq = new int[26];
        int deletions = 0;
        for (char ch : s.toCharArray())
            freq[ch - 'a']++;

        Set<Integer> mySet = new HashSet<>();

        for (int i = 0; i < 26; ++i) {
            while (freq[i] > 0 && mySet.contains(freq[i])) {
                freq[i]--;
                deletions++;
            }
            mySet.add(freq[i]);
        }
        return deletions;
    }
}

public class MinimumDeletionstoMakeCharacterFrequenciesUnique {
    public static void main(String[] args) {
        String s = "ceabaacb";
        System.out.println(new Solution().minDeletions(s));
    }
}

File: Year-2023/LeetCode-September-challenge/MinimumDeletionstoMakeCharacterFrequenciesUnique.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minDeletions(string s) {
        int freq[26] = {0}, deletions = 0;
        for(auto ch: s) freq[ch - 'a']++;

        set<int> mySet;
        for(int i = 0; i < 26; ++i){
            while(freq[i] > 0 && !mySet.insert(freq[i]).second){
                freq[i]--;
                deletions++;
            }
        }
        return deletions;
    }
};

int main(){
    string s = "ceabaacb";
    Solution *S = new Solution();
    cout<<S->minDeletions(s)<<endl;
    return 0;
}